crates/ingress-dispatcher/src/lib.rs

```diff
@@ -310,56 +282,38 @@ impl IngressDispatcherInput {
         IngressDispatcherInput::DedupMessageAck(dedup_name, seq_number)
     }
 
-    pub fn response(response: IngressResponseMessage) -> Self {
+    pub fn response(response: IngressResponse) -> Self {
         IngressDispatcherInput::Response(response)
     }
 }
 
-#[derive(Debug)]
-pub enum IngressDispatcherOutput {
-    Envelope(Envelope),
-    Ack(AckResponse),
-}
-
-#[derive(Debug)]
-pub struct AckResponse {
-    pub shuffle_target: PeerId,
-    pub kind: AckKind,
-}
-
-impl IngressDispatcherOutput {
-    pub fn service_invocation(
-        service_invocation: ServiceInvocation,
-        from_node_id: GenerationalNodeId,
-        deduplication_source: Option<String>,
-        msg_index: MessageIndex,
-    ) -> Self {
-        let ack_mode = if deduplication_source.is_some() {
-            AckMode::Dedup
-        } else {
-            AckMode::Ack
-        };
-
-        let header = Header {
-            source: Source::Ingress {
-                node_id: from_node_id,
-                sequence_number: msg_index,
-                dedup_key: deduplication_source,
-                // todo: Retrieve proper node config version
-                nodes_config_version: Version::MIN,
-            },
-            dest: Destination::Processor {
-                partition_key: service_invocation.fid.partition_key(),
-            },
-            ack_mode,
-        };
-
-        Self::Envelope(Envelope::new(header, Command::Invoke(service_invocation)))
-    }
-
-    pub fn shuffle_ack(ack_response: AckResponse) -> Self {
-        Self::Ack(ack_response)
-    }
+pub fn wrap_service_invocation_in_envelope(
+    service_invocation: ServiceInvocation,
+    from_node_id: GenerationalNodeId,
+    deduplication_source: Option<String>,
+    msg_index: MessageIndex,
+) -> Envelope {
+    let ack_mode = if deduplication_source.is_some() {
+        AckMode::Dedup
+    } else {
+        AckMode::Ack
+    };
+
+    let header = Header {
+        source: Source::Ingress {
+            node_id: from_node_id,
+            sequence_number: msg_index,
+            dedup_key: deduplication_source,
+            // todo: Retrieve proper node config version
+            nodes_config_version: Version::MIN,
```

Comment:
This is now available in `metadata().nodes_config_version()` if you want to use it.

Reply:
Cool, will update it :-)